tests/load_balancer_test.go

```diff
@@ -0,0 +1,280 @@
+			ExpectSuccess:     true,
+			ExpectedDocuments: []interface{}{collectionData["users"][2], collectionData["users"][0], collectionData["users"][5]},
+			DocumentType:      reflect.TypeOf(UserDoc{}),
+		},
+		queryTest{
+			Query:         "FOR u IN users FILTER u.age<@maxAge SORT u.name RETURN u",
+			BindVars:      map[string]interface{}{"maxage": 20},
+			ExpectSuccess: false, // `@maxage` versus `@maxAge`
+		},
+		queryTest{
+			Query:             "FOR u IN users SORT u.age RETURN u.age",
+			ExpectedDocuments: []interface{}{12, 12, 13, 25, 42, 67},
+			DocumentType:      reflect.TypeOf(12),
+			ExpectSuccess:     true,
+		},
+		queryTest{
+			Query:             "FOR p IN users COLLECT a = p.age WITH COUNT INTO c SORT a RETURN [a, c]",
+			ExpectedDocuments: []interface{}{[]int{12, 2}, []int{13, 1}, []int{25, 1}, []int{42, 1}, []int{67, 1}},
+			DocumentType:      reflect.TypeOf([]int{}),
+			ExpectSuccess:     true,
+		},
+		queryTest{
+			Query:             "FOR u IN users SORT u.name RETURN u.name",
+			ExpectedDocuments: []interface{}{"Blair", "Clair", "Jake", "John", "Johnny", "Zz"},
+			DocumentType:      reflect.TypeOf("foo"),
+			ExpectSuccess:     true,
+		},*/
+	}
+
+	var r driver.Response
+	// Setup context alternatives
+	contexts := []queryTestContext{
+		/*queryTestContext{driver.WithResponse(nil, &r), false},
+		queryTestContext{driver.WithResponse(context.Background(), &r), false},
+		queryTestContext{driver.WithResponse(driver.WithQueryCount(nil), &r), true},
+		queryTestContext{driver.WithResponse(driver.WithQueryCount(nil, true), &r), true},
+		queryTestContext{driver.WithResponse(driver.WithQueryCount(nil, false), &r), false},*/
+		queryTestContext{driver.WithResponse(driver.WithQueryBatchSize(nil, 1), &r), false},
+		/*queryTestContext{driver.WithResponse(driver.WithQueryCache(nil), &r), false},
+		queryTestContext{driver.WithResponse(driver.WithQueryCache(nil, true), &r), false},
+		queryTestContext{driver.WithResponse(driver.WithQueryCache(nil, false), &r), false},
+		queryTestContext{driver.WithResponse(driver.WithQueryMemoryLimit(nil, 60000), &r), false},
+		queryTestContext{driver.WithResponse(driver.WithQueryTTL(nil, time.Minute), &r), false},
+		queryTestContext{driver.WithResponse(driver.WithQueryBatchSize(driver.WithQueryCount(nil), 1), &r), true},
+		queryTestContext{driver.WithResponse(driver.WithQueryCache(driver.WithQueryCount(driver.WithQueryBatchSize(nil, 2))), &r), true},*/
+	}
+
+	// keep track of whether at least one request was forwarded internally to the
+	// correct coordinator behind the load balancer
+	someRequestForwarded := false
+
+	// Run tests for every context alternative
+	for _, qctx := range contexts {
+		ctx := qctx.Context
+		for i, test := range tests {
+			cursor, err := db.Query(ctx, test.Query, test.BindVars)
+			if err == nil {
+				// Close upon exit of the function
+				defer cursor.Close()
+			}
+			if test.ExpectSuccess {
```

Comment:
This entire file needs `go fmt`. Since most editors do that automatically, any future change would cause a huge diff if you do not format now.